### PR TITLE
token for recover password and reset password

### PR DESCRIPTION
**Pull request by ps-eqsdigital**
recover password - user receives token in email
reset password - user gives token and new password

### CODE REVIEW

File: api/DataAccess/DataAccessObjects/UserDataAccessObject.cs [hunk 1]
```diff
@@ -64,5 +64,10 @@ public async Task<UserTokenAuthentication> GetTokenUuidByToken(string token)
             var result = await _context.Set<UserTokenAuthentication>().Where(x => x.Token == token).ToListAsync();
             return result.Where(x => !x.IsDeleted).SingleOrDefault()!;
         }
+        public async Task<User> GetUserByPasswordResetToken(string token)
+        {
+            var result = await _context.Set<User>().Where(x=>x.PasswordResetToken == token).ToListAsync();
```
Comment (kripton69) [inline api/DataAccess/DataAccessObjects/UserDataAccessObject.cs]: types

File: api/api/Controllers/UserController.cs [hunk 2]
```diff
@@ -81,6 +81,26 @@ public async Task<ActionResult> Login(string email, string password)
...
+        [HttpPost("recoverPassword")]
+        public async Task<ActionResult> RecoverPassword(string email)
+        {
+            OperationResult result = await _userBusinessObject.RecoverPassword(email);
+            if (result.Exception is Exception)
+            {
+                return StatusCode(400);
+            }
+            return Ok(result);
+        }
```
Comment (kripton69) [inline api/api/Controllers/UserController.cs:95]: break line

File: api/DataAccess/DataAccessObjects/UserDataAccessObject.cs (hunk 1 above)
Comment (ps-eqsdigital) [reply api/DataAccess/DataAccessObjects/UserDataAccessObject.cs]: done

File: api/api/Controllers/UserController.cs (hunk 2 above)
Comment (ps-eqsdigital) [reply api/api/Controllers/UserController.cs:95]: done

File: api/Business/Utils/TokenUtils.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json")
+            .Build();
+
+            List<Claim> claims = new List<Claim>
+            {
+                new Claim(ClaimTypes.Name, user.Name!),
+            };
+
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(configuration.GetSection("AppSettings:Token").Value!));
```
Comment (kripton69) [inline api/Business/Utils/TokenUtils.cs]: types

Comment (ps-eqsdigital) [reply api/Business/Utils/TokenUtils.cs]: done

**Review by kripton69 — APPROVED** [review]

**Pull request merged**